### ISSUE TITLE
AbstractJoiner NullPointerException

### ISSUE THREAD

**Issue by darbie** [hazelcast/hazelcast#12676]
I was running a 2 node system.  Both nodes on windows with version 3.8 of hazelcast. I am using java 1.8. I do not know how to recreate this.  The system is normally fine, but somehow got into this state. 

On node1 I see the following in the log:
```java
INFO  [2018-03-19 08:33:01.611] [hz._hzInstance_1_vmscore_1-bc073b4f-1636-4b24-b0e3-3b0d747db533.priority-generic-operation.thread-0] com.hazelcast.internal.cluster.impl.operations.SplitBrainMergeValidationOperation: [192.168.3.4]:5701 [vmscore_1-bc073b4f-1636-4b24-b0e3-3b0d747db533] [3.8] Join check from [192.168.3.5]:5701 failed validation due to incompatible version,remote cluster version is null, this cluster is 3.8
```

On node2 I see the following in the log:
```java
WARN  [2018-03-19 08:32:59.157] [hz._hzInstance_1_vmscore_1-bc073b4f-1636-4b24-b0e3-3b0d747db533.cached.thread-8] com.hazelcast.internal.cluster.impl.MulticastJoiner: [192.168.3.5]:5701 [vmscore_1-bc073b4f-1636-4b24-b0e3-3b0d747db533] [3.8] null 

java.lang.NullPointerException: null
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.checkCompatibleSplitBrainJoinMessage(AbstractJoiner.java:294)
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.shouldMerge(AbstractJoiner.java:234)
	at com.hazelcast.internal.cluster.impl.MulticastJoiner.searchForOtherClusters(MulticastJoiner.java:131)
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75)
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42)
```

**Comment by mmedenjak** [comment]
Hi @darbie ! 
Seems the node2 doesn't have it's cluster version set, causing the failed check on node1 and the `NullPointerException` on node2. Do you still have the full logs from node2? Can you post them here?

**Comment by mmedenjak** [comment]
Can you try upgrading to hazelcast 3.8.1, 3.9 or 3.10? Seems your issue might have been fixed in 3.8.1 - https://github.com/hazelcast/hazelcast/pull/9946

**Comment by mdogan** [comment]
Hi @darbie,

It looks like this issue was fixed in 3.8.1. If you still observe the issue after upgrading to 3.8.1 or a newer version, please re-open.
